### DISCUSSION TITLE
Template library search and IA

### DISCUSSION THREAD

**Discussion by ackernaut** [ParabolInc/parabol#6249] [Ideas]
Our current template navigator allows folks to browse templates created by their team, shared across their org, and public templates. In the public templates I’m counting 38 retro templates. The sort order is not apparent. I have to scroll over 6 times to get to the last templates (the list shows 6 templates at a time, filling the dialog height at the larger breakpoints).

### Add an inline search to our template viewer?

In the short term I could see a quick inline search being useful so folks can find that template they know they’re looking for. If it makes sense we could open a design issue.

### Rambing on fuzzy, future possibilities

In the future, I imagine a place where we have more activity types and could normalize some patterns for exploring, selecting, editing, and building your own. We can cross that bridge when we get there. I could imagine not only picking from templates by meeting type, but maybe even templated phases and breakout activities. When building your own meeting, when adding a specific phase activity what options do you have for the phase, maybe some things are templated?

So my questions in the future would be:

- What kind of activities (complete meetings or editable phases) might start from a template of options? Decision making? Brain storming? Answering a prompt? 5 whys? What about a templated onboarding or sprint cadence?
- How might we lay these templated experiences out in way that is easy to navigate, easy to edit and make your own, or build from scratch? (We could probably learn a thing or 2 from Enrique’s latest design on a start meeting dialog)

**Comment by ackernaut** [comment]
Has anybody given us feedback on this? I haven’t heard anything

**Comment by jordanh** [comment]
I feel this tension!

**Comment by nickoferrall** [comment]
I completely agree! @garethaledavies created an issue around this tension too: https://github.com/ParabolInc/parabol/issues/5972

**Comment by avivapinchas** [comment]
I like the idea of a search field! 

🤔  I imagine that'll be helpful for the super common types of templates, but for folks looking for more general inspiration, they won't know where to search. I like the idea of some kind of tags to indicate something like use case. Examples: good for end-of-year, good for new teams, good for pre-mortems, etc. 

> The sort order is not apparent.

I think the most useful option would be to sort by popularity. Or perhaps we just define a specific sort order for now, based on a combo of making common ones quick-to-find, and highlighting some goodies?
